.github/workflows/docbuild.yml

```diff
@@ -0,0 +1,45 @@
+# Name your workflow.
+name: Deploy DocC
+on:
+  # Runs on pushes targeting the default branch
+  push:
+    branches: ["main"]
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  # Single deploy job since we're just deploying
+  deploy:
+    environment:
+      # Must be set to this for deploying to GitHub Pages
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    runs-on: macos-12
+    steps:
+      - name: Checkout 🛎️
+        uses: actions/checkout@v3
+      - name: Build DocC
+        run: |
+          xcodebuild docbuild -scheme ACKategories \
```

Comment:
As long as you use xcodebuild I would call the load xcode version action as all other actions do to make sure you use correct Xcode version